agent/consul/server.go

```diff
@@ -380,21 +380,20 @@ func NewServer(config *Config, flat Deps, publicGRPCServer *grpc.Server) (*Serve
 	recorder := middleware.NewRequestRecorder(serverLogger)
 	// Create server.
 	s := &Server{
-		config:       config,
-		tokens:       flat.Tokens,
-		connPool:     flat.ConnPool,
-		grpcConnPool: flat.GRPCConnPool,
-		eventChLAN:   make(chan serf.Event, serfEventChSize),
-		eventChWAN:   make(chan serf.Event, serfEventChSize),
-		logger:       serverLogger,
-		loggers:      loggers,
-		leaveCh:      make(chan struct{}),
-		reconcileCh:  make(chan serf.Member, reconcileChSize),
-		router:       flat.Router,
-		rpcRecorder:  recorder,
-		// TODO(rpc-metrics-improv): consider pulling out the interceptor from config in order to isolate testing
-		rpcServer:               rpc.NewServerWithOpts(rpc.WithServerServiceCallInterceptor(middleware.GetNetRPCInterceptor(recorder))),
-		insecureRPCServer:       rpc.NewServerWithOpts(rpc.WithServerServiceCallInterceptor(middleware.GetNetRPCInterceptor(recorder))),
+		config:                  config,
+		tokens:                  flat.Tokens,
+		connPool:                flat.ConnPool,
+		grpcConnPool:            flat.GRPCConnPool,
+		eventChLAN:              make(chan serf.Event, serfEventChSize),
+		eventChWAN:              make(chan serf.Event, serfEventChSize),
+		logger:                  serverLogger,
+		loggers:                 loggers,
+		leaveCh:                 make(chan struct{}),
+		reconcileCh:             make(chan serf.Member, reconcileChSize),
+		router:                  flat.Router,
+		rpcRecorder:             recorder,
+		rpcServer:               rpc.NewServer(),
```

Comment:
What are the implication of creating a new rpc server in here and then creating a new one later, do the NewServer() call use any system resources (ex socket)?
I think another way in here is to make `GetNetRPCInterceptorFunc` return a nil (instead of being nil) in the case we don't want to intercept the calls and make sure that the nil interceptor is a no op.